token-metadata/program/src/instruction/metadata.rs

```diff
@@ -100,20 +100,278 @@ pub enum UpdateArgs {
     },
+    UpdateAuthorityV2 {
+        /// The new update authority.
+        new_update_authority: Option<Pubkey>,
+        /// The metadata details.
+        data: Option<Data>,
+        /// Indicates whether the primary sale has happened or not (once set to `true`, it cannot be
+        /// changed back).
+        primary_sale_happened: Option<bool>,
+        // Indicates Whether the data struct is mutable or not (once set to `true`, it cannot be
+        /// changed back).
+        is_mutable: Option<bool>,
+        /// Collection information.
+        collection: CollectionToggle,
+        /// Additional details of the collection.
+        collection_details: CollectionDetailsToggle,
+        /// Uses information.
+        uses: UsesToggle,
+        // Programmable rule set configuration (only applicable to `Programmable` asset types).
+        rule_set: RuleSetToggle,
+        /// Token standard.
+        token_standard: Option<TokenStandard>,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    AuthorityItemDelegateV2 {
+        /// The new update authority.
+        new_update_authority: Option<Pubkey>,
+        /// Indicates whether the primary sale has happened or not (once set to `true`, it cannot be
+        /// changed back).
+        primary_sale_happened: Option<bool>,
+        // Indicates Whether the data struct is mutable or not (once set to `true`, it cannot be
+        /// changed back).
+        is_mutable: Option<bool>,
+        /// Token standard.
+        token_standard: Option<TokenStandard>,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    CollectionDelegateV2 {
+        /// Collection information.
+        collection: CollectionToggle,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    DataDelegateV2 {
+        /// The metadata details.
+        data: Option<Data>,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    ProgConfigDelegateV2 {
+        // Programmable rule set configuration (only applicable to `Programmable` asset types).
+        rule_set: RuleSetToggle,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    DataItemDelegateV2 {
+        /// The metadata details.
+        data: Option<Data>,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    CollectionItemDelegateV2 {
+        /// Collection information.
+        collection: CollectionToggle,
+        /// Required authorization data to validate the request.
+        authorization_data: Option<AuthorizationData>,
+    },
+    ProgConfigItemDelegateV2 {
```

Comment:
I am not a big fan of abbreviated names. If we can drop the `Delegate` from the name, this one could be names `ProgrammableConfigItem`.

Reply:
Dropped the abbreviation and put in the full "Programmable".  See other comment for more info.